File: search/binary_search.py
size = int(input("Enter number of elements in the list:"))
li = []
print("Enter elements one by one:")
for x in range (0,size):
	li.append(int(input()))
etf = int(input("Enter element to find: "))
li.sort()
min = 0
max = size-1
found = False
while(min<=max):
	mid = (min+max)//2
	if(li[mid] == etf):
		print("Element found at "+str(mid))
		found = True
		break
	elif(etf>li[mid]):
		min = mid+1
	else:
		max = mid-1
if(not found):
	print("Element not found")

File: Binary Search/binary_search.cs
using System;

public class BinarySearch
{
    public static void Main(string[] args)
    {
        Console.WriteLine("Enter the range of the sorted array: ");
        int n = int.Parse(Console.ReadLine());
        Console.WriteLine("Enter elements of the sorted array (one-line):");
        int[] ar = new int[n];
        string[] ip = Console.ReadLine().Split(' ');
        for(int i = 0; i < n; i++)
            ar[i] = int.Parse(ip[i]);
        Console.WriteLine("Enter the element to be searched: ");
        int x = int.Parse(Console.ReadLine());
        int beg = 0, end = n-1, f = 0;
        while(beg <= end)
        {
            int mid = beg + (end - beg) / 2;
            if(ar[mid] == x)
            {
                f = 1;
                Console.WriteLine($"Element found at {mid+1} position.");
                break;
            }
            else if(ar[mid] < x)
            {
                beg = mid + 1;
            }
            else
            {
                end = mid - 1;
            }
        }
        if(f == 0)
            Console.WriteLine("Element not found!");
    }
}